charts/k3k/templates/deployment.yaml

```
@@ -15,25 +21,68 @@ spec:
       labels:
         {{- include "k3k.selectorLabels" . | nindent 8 }}
     spec:
+      {{- with (coalesce .Values.controller.imagePullSecrets .Values.imagePullSecrets)}}
+      imagePullSecrets:
+        {{- $imagePullSecrets := list }}
+            {{- range . }}
+                {{- if kindIs "string" . }}
+                    {{- $imagePullSecrets = append $imagePullSecrets (dict "name" .) }}
+                {{- else }}
+                    {{- $imagePullSecrets = append $imagePullSecrets . }}
+            {{- end }}
+        {{- end }}
+        {{- toYaml $imagePullSecrets | nindent 8 }}
+        {{- end }}
       containers:
-        - image: "{{ .Values.image.repository }}:{{ default .Chart.AppVersion .Values.image.tag }}"
-          imagePullPolicy: {{ .Values.image.pullPolicy }}
+        - image: "{{- $controllerRegistry }}/{{ .Values.controller.image.repository }}:{{ default .Chart.AppVersion .Values.controller.image.tag }}"
```

I know that it will be a bit long, but probably it's better to have it inline for clarity, wdyt?

```suggestion
        - image: "{{- default .Values.imageRegistry .Values.controller.image.registry }}/{{ .Values.controller.image.repository }}:{{ default .Chart.AppVersion .Values.controller.image.tag }}"
```